src/typescript-worker/typescript.ts

```diff
@@ -37,11 +38,27 @@ export class TypeScriptBuilder {
   private readonly _cdn: CachingCdn;
   private readonly _importMapResolver: ImportMapResolver;
 
+  private _languageServiceHost: WorkerLanguageServiceHost | undefined;
+  private _languageService: LanguageService | undefined;
+
   constructor(cdn: CachingCdn, importMapResolver: ImportMapResolver) {
     this._cdn = cdn;
     this._importMapResolver = importMapResolver;
   }
 
+  _createLanguageService() {
+    // Initialize the language service and the host to manage it
+    // Files are added on the fly
+    this._languageServiceHost = new WorkerLanguageServiceHost(
```

Comment:
Maybe simpler to do this initialization on construction using a class field initializer? E.g.:

```ts
  private readonly _languageServiceHost = new WorkerLanguageServiceHost(
    self.origin,
    compilerOptions
  );

  private readonly _languageService = ts.createLanguageService(
    this._languageServiceHost,
    ts.createDocumentRegistry()
  );
```

Reply:
You're totally right. I did pass the files into the function earlier, but as I removed that and moved the file passing elsewhere, this can now be done in the constructor.